src/Flysystem/FlysystemAssetStore.php

```diff
@@ -597,10 +598,16 @@ function (ParsedFileID $parsedFileID, Filesystem $fs, FileResolutionStrategy $st
                             // Move cached hash value to new location
                             $hasher->move($origin, $fs, $destination);
                         }
-                        $this->truncateDirectory(dirname($origin ?? ''), $fs);
+                        $tryTruncate[dirname($origin)] = true;
                     }
                 }
 
+                // Truncate any directories that held removed files.
+                // We do this all in one go like this to reduce duplicate calls which can reduce performance.
+                foreach (array_keys($tryTruncate) as $dir) {
+                    $this->truncateDirectory($dir, $fs);
+                }
```

Comment:
```suggestion
                $this->truncateDirectories($dirsToTruncate);
```

Update for all other examples below


And create a new private method

```php
private function truncateDirectories(array $dirs): void
{
    $dirs = array_unique($dirs);
    foreach ($dirs as $dir) {
        $this->truncateDirectory($dir, $fs);
    }
}
```

Reply:
I'd prefer not to do this. I don't see any particular advantage.